terragrunt.hcl.secure.example

```diff
@@ -18,6 +18,7 @@ terraform {
 
 inputs = {
     projectPrefix          = "prefixforassets"
+    resourceOwner          = "lastname"
```

Comment:
can an email address be used for this value? If so, you may consider using the useremail value in the secret vars

Reply:
I went with lastname since email cannot be used on all 3 CSPs. Google for example does not allow email format.

GCP error
"Only hyphens (-), underscores (_), lowercase characters, and numbers are allowed. International characters are allowed."

Options:
1. Use email for CSPs that support, and lastname for GCP
2. Or...use lastname for all CSP

Preference?

Reply:
FWIW I suggest we standardise on whatever is deemed most informative and deal with CSP variations in those modules. Came across a similar situation when testing k8s site meshes with @piyerf5 and a brute force remediation is a one liner to add to Google modules, say. E.g. https://github.com/memes/f5xc-gke-site-mesh/blob/main/f5xc-site/main.tf#L26

Reply:
A more sophisticated regex could transform emails if needed, and validators added to the label variable to enforce at least owner = email at the top level

Reply:
Last name is going to be an issue in larger orgs: smith, jones, wang, kim, etc.

Reply:
so would first initial hyphen last name be the solution to work for all CSPs based upon what we know today?

Reply:
The value is arbitrary, but should link to the deployer for tagging and "who did it" purposes. It can be email, last name, unique identifier, business unit. Code-wise though, it is applied to the resources in the tag lines with key "Owner" to satisfy IT requirements of having an "Owner" tag. Currently, the objects across various modules have a "Name" tag or no tag at all in some resources. I chose 'resourceOwner' since it was already in code that I copied in. I can very well use another existing input and simply apply that value to the "Owner" tag.

Maybe further discussion...for now I'm going to make it work with my current change.

Reply:
I understand and accept the constraint.